core/cmd/app.go

```diff
@@ -50,8 +52,34 @@ func NewApp(client *Client) *cli.App {
 			Name:  "insecure-skip-verify",
 			Usage: "optional, applies only in client mode when making remote API calls. If turned on, SSL certificate verification will be disabled. This is mostly useful for people who want to use Chainlink with a self-signed TLS certificate",
 		},
+		cli.StringFlag{
+			Name:   "config, c",
+			Usage:  "TOML configuration file via flag, or raw TOML via env var. If used, legacy env vars must not be set.", //TODO
+			EnvVar: "CL_CONFIG",
+		},
 	}
 	app.Before = func(c *cli.Context) error {
+		if c.IsSet("config") {
+			// TOML
+			configTOML := os.Getenv("CL_CONFIG")
+			if configTOML == "" {
+				fileName := c.String("config")
+				b, err := os.ReadFile(fileName)
+				if err != nil {
+					return errors.Wrapf(err, "failed to read config file: %s", fileName)
+				}
+				configTOML = string(b)
+			}
```

Comment:
Can you put this logic into a function so it could be easily reused for secrets and runtime config? Maybe even together with unmarshalling into a given struct? app.go already depends on v2 types anyway so it might just pass them to NewConfig directly instead of TOML strings (assuming this is the way we want it to go).

Reply:
I'm not sure I'm following. This is already shared between all commands and it seems logical to add secrets here too. We shouldn't need to duplicate much of anything

Reply:
I was thinking about something along the lines of

```
func ParseConfigFromFile(fileName string, destConfig interface{}) error {
	tomlBytes, err := os.ReadFile(fileName)
	if err != nil {
		return errors.Wrapf(err, "failed to read config file: %s", fileName)
	}
	err = toml.Unmarshal(tomlBytes, destConfig)
	return err
}
```

Reply:
Hmmm maybe, but the env differences with secrets might be too different 🤷